src/storage/src/engine.rs

```diff
@@ -492,23 +498,35 @@ impl<S: LogStore> EngineInner<S> {
 
 #[cfg(test)]
 mod tests {
-    use common_test_util::temp_dir::create_temp_dir;
+    use std::ffi::OsStr;
+    use std::path::Path;
+
+    use common_test_util::temp_dir::{create_temp_dir, TempDir};
     use datatypes::type_id::LogicalTypeId;
+    use datatypes::vectors::{Float32Vector, Int32Vector, TimestampMillisecondVector, VectorRef};
+    use log_store::raft_engine::log_store::RaftEngineLogStore;
     use log_store::test_util::log_store_util;
     use object_store::services::Fs;
-    use store_api::storage::Region;
+    use store_api::storage::{FlushContext, Region, WriteContext, WriteRequest};
 
     use super::*;
     use crate::compaction::noop::NoopCompactionScheduler;
     use crate::test_util::descriptor_util::RegionDescBuilder;
 
-    #[tokio::test]
-    async fn test_create_new_region() {
+    type TestEngine = EngineImpl<RaftEngineLogStore>;
+    type TestRegion = RegionImpl<RaftEngineLogStore>;
+
+    async fn create_engine_and_region(
+        tmp_dir: &TempDir,
+        region_name: &str,
+        region_id: u64,
+        ctx: &EngineContext,
+    ) -> (TestEngine, TestRegion) {
         let log_file_dir = create_temp_dir("test_engine_wal");
```

Comment:
We should pass the log_file_dir as an argument so the temp dir won't be dropped after this function returns.